aeppl/transforms.py

```diff
@@ -579,17 +591,31 @@ def log_jac_det(self, value, *inputs):
         return -at.log(value)
 
 
-class ReciprocalTransform(RVTransform):
-    name = "reciprocal"
+class PowerTransform(RVTransform):
+    name = "power"
+
+    def __init__(self, transform_args_fn):
+        self.transform_args_fn = transform_args_fn
 
     def forward(self, value, *inputs):
-        return at.reciprocal(value)
+        power = self.transform_args_fn(*inputs)
+        return at.power(value, power)
 
     def backward(self, value, *inputs):
-        return at.reciprocal(value)
+        power = self.transform_args_fn(*inputs)
+
+        inv_power = at.reciprocal(power)
+        return at.switch(
+            at.eq(at.mod(power, 2), 0),
+            at.power(value, inv_power),
+            at.sgn(value) * at.power(at.abs(value), inv_power),
+        )
 
     def log_jac_det(self, value, *inputs):
-        return -2 * at.log(value)
+        from aeppl.logprob import xlogy0
+
+        power = self.transform_args_fn(*inputs)
```

Comment:
Don't you need to use `inv_power = at.reciprocal(power)` instead of `power` in the expression of the jacobian?